pymeasure/adapters/adapter.py

```diff
@@ -41,37 +43,75 @@ class Adapter:
     def __init__(self, preprocess_reply=None, **kwargs):
         self.preprocess_reply = preprocess_reply
         self.connection = None
+        self.log = logging.Logger("Adapter")
+        self.log.addHandler(logging.NullHandler())
 
     def __del__(self):
         """Close connection upon garbage collection of the device"""
         if self.connection is not None:
             self.connection.close()
 
+    # Directly called methods.
```

Comment:
we will need a note in the changelog that we now do more in read/write/..., and people that have implemented custom adapters that are not in pymeasure need to change their overrides.